cleanlab/experimental/label_issues_batched.py

```diff
@@ -0,0 +1,461 @@
+>>> i = 0
+>>> while i < n:
+>>>     end_index = i + batch_size
+>>>     labels_batch = labels[i:end_index]
+>>>     pred_probs_batch = pred_probs[i:end_index,:]
+>>>     i = end_index
+>>>     batch_results = lab.score_label_quality(labels_batch, pred_probs_batch)
+>>> # Indices of examples with label issues, sorted by label quality score (most severe to least severe):
+>>> indices_of_examples_with_issues = lab.get_label_issues()
+>>> # If your `pred_probs` and `labels` are arrays already in memory,
+>>> # then you can use this shortcut for all of the above:
+>>> indices_of_examples_with_issues = find_label_issues_batched(labels, pred_probs, batch_size=10000)
+"""
+
+import numpy as np
+from typing import Optional, List
+
+from cleanlab.count import get_confident_thresholds
+from cleanlab.rank import get_label_quality_scores, find_top_issues
+from cleanlab.typing import LabelLike
+from cleanlab.internal.util import value_counts_fill_missing_classes
+
+
+EPS = 1e-6  # small number
+
+
+class LabelInspector:
+    """
+    Class for finding label issues in big datasets where memory becomes a problem for other cleanlab methods.
+    Only create one such object per dataset and do not try to use the same ``LabelInspector`` across 2 datasets.
+    For efficiency, this class does little input checking.
+    You can first run :py:func:`filter.find_label_issues <cleanlab.filter.find_label_issues>`
+    on a small subset of your data to verify your inputs are properly formatted.
+    Do NOT modify any of the attributes of this class yourself!
+    Multi-label classification is not supported by this class, it is only for multi-class classification.
+
+    Parameters
+    ----------
+    num_class : int
+      The number of classes in your multi-class classification task.
+
+    store_results : bool, optional
+      Whether this object will store all label quality scores.
+      Set this to False if you encounter memory problems even for small batch sizes (~1000).
+      If ``False``, you can still identify the label issues yourself by aggregating
+      the label quality scores for each batch, sorting them across all batches, and returning the top ``T`` indices
+      with ``T = self.get_num_issues()``.
+
+    verbose : bool, optional
+      Whether to suppress print statements or not.
+
+    quality_score_kwargs : dict, optional
+      Keyword arguments to pass into :py:func:`rank.get_label_quality_scores <cleanlab.rank.get_label_quality_scores>`.
+
+    num_issue_kwargs : dict, optional
+      Keyword arguments to :py:func:`count.num_label_issues()` <cleanlab.count.num_label_issues>`
+      to control estimation of the number of label issues.
+      The only supported kwarg here for now is: `estimation_method`.
+    """
+
+    def __init__(
+        self,
+        *,
+        num_class: int,
+        store_results: bool = True,
+        verbose: bool = True,
+        quality_score_kwargs: Optional[dict] = None,
```

Comment:
Why not do below as default args? This way you don't need the first two if statements
```
def __init__(
   self,
   *,
   num_class: int,
   ...
   quality_score_kwargs: dict = {},
   num_issues_kwargs: dict = {},
):
```

Reply:
you are never supposed to empty dicts as default args, core Python principle

https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python